lib/class-wp-rest-block-editor-settings-controller.php

```diff
@@ -0,0 +1,267 @@
+	 *
+	 * @since 5.8.0
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		$this->schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'block-editor-settings-item',
+			'type'       => 'object',
+			'properties' => array(
+				'__unstableEnableFullSiteEditingBlocks'  => array(
+					'description' => __( 'Enables experimental Full Site Editing blocks', 'gutenberg' ),
+					'type'        => 'boolean',
+					'context'     => array( 'post-editor', 'site-editor', 'widgets-editor' ),
+				),
+
+				'styles'                                 => array(
+					'description' => __( 'Editor styles', 'gutenberg' ),
+					'type'        => 'array',
+					'context'     => array( 'post-editor', 'site-editor', 'widgets-editor' ),
+				),
+
+				'supportsTemplateMode'                   => array(
+					'description' => __( 'Returns if the current theme is full site editing-enabled or not.', 'gutenberg' ),
+					'type'        => 'boolean',
+					'context'     => array( 'post-editor', 'site-editor', 'widgets-editor' ),
+				),
+
+				'supportsLayout'                         => array(
+					'description' => __( 'Enable/disable layouts support in container blocks.', 'gutenberg' ),
+					'type'        => 'boolean',
+					'context'     => array( 'post-editor', 'site-editor', 'widgets-editor' ),
+				),
+
+				'widgetTypesToHideFromLegacyWidgetBlock' => array(
+					'description' => __( 'Widget types to hide from Legacy Widget block.', 'gutenberg' ),
+					'type'        => 'array',
+					'context'     => array( 'post-editor', 'site-editor', 'widgets-editor' ),
+				),
+
+				'__experimentalFeatures'                 => array(
+					'description' => __( 'Active theme settings and default values.', 'gutenberg' ),
+					'type'        => 'object',
+					'context'     => array( 'post-editor', 'site-editor', 'widgets-editor' ),
+				),
+
+				'__experimentalGlobalStylesUserEntityId' => array(
+					'description' => __( 'Global styles user entity ID.', 'gutenberg' ),
+					'type'        => 'integer',
+					'context'     => array( 'site-editor' ),
+				),
+
+				'__experimentalGlobalStylesBaseStyles'   => array(
+					'description' => __( 'Global styles settings.', 'gutenberg' ),
+					'type'        => 'object',
+					'context'     => array( 'site-editor' ),
+				),
+
+				'alignWide'                              => array(
+					'description' => __( 'Enable/Disable Wide/Full Alignments.', 'gutenberg' ),
+					'type'        => 'boolean',
+					'context'     => array( 'post-editor', 'site-editor', 'widgets-editor' ),
+				),
+
+				'allowedBlockTypes'                      => array(
+					'description' => __( 'List of allowed block types.', 'gutenberg' ),
+					'type'        => 'boolean',
+					'context'     => array( 'post-editor', 'site-editor', 'widgets-editor' ),
```

Comment:
I've noted this in other PRs and on Trac, but the lack of post type context constitutes a large functional break for all of the Gutenberg sites I've worked on, and I expect for many others as well. Being able to restrict certain blocks to certain post types is a crucial feature of many advanced block editor implementations.

Reply:
We discussed with @chrisvanpatten in https://github.com/WordPress/wordpress-develop/pull/1118#discussion_r627396445 that it should be considered passing additional params in addition to the context to bring feature parity with how the edit post editor works today. It looks like the possibility to change settings per individual post/page is used by the community. It means that we should explore how params like `post_id` could be passed and consumed by this endpoint.